prost-build/src/code_generator.rs

```diff
@@ -322,73 +322,65 @@ impl CodeGenerator<'_> {
         let prost_path = self.config.prost_path.as_deref().unwrap_or("::prost");
         let string_path = format!("{prost_path}::alloc::string::String");
 
-        let full_name = format!(
-            "{}{}{}{}{message_name}",
-            self.package.trim_matches('.'),
-            if self.package.is_empty() { "" } else { "." },
-            self.type_path.join("."),
-            if self.type_path.is_empty() { "" } else { "." },
-        );
+        let full_name = FullyQualifiedName::new(&self.package, &self.type_path, message_name);
         let domain_name = self
             .config
             .type_name_domains
             .get_first(fq_message_name)
             .map_or("", |name| name.as_str());
 
+        let full_name_str = full_name.as_ref().trim_start_matches('.');
         self.buf.push_str(&format!(
-            r#"fn full_name() -> {string_path} {{ "{full_name}".into() }}"#,
+            r#"fn full_name() -> {string_path} {{ "{}".into() }}"#,
+            full_name_str,
         ));
 
         self.buf.push_str(&format!(
-            r#"fn type_url() -> {string_path} {{ "{domain_name}/{full_name}".into() }}"#,
+            r#"fn type_url() -> {string_path} {{ "{domain_name}/{}".into() }}"#,
+            full_name_str
```

Comment:
I prefer the original with variable names in the text.
```suggestion
            r#"fn type_url() -> {string_path} {{ "{domain_name}/{full_name_str}".into() }}"#,
```